src/libraries/Common/src/System/Security/Cryptography/CompositeMLDsaManaged.cs

```diff
@@ -0,0 +1,682 @@
+
+            if (_componentAlgorithm.TryExportPublicKey(destination.Slice(AlgorithmDetails.MLDsaAlgorithm.PublicKeySizeInBytes), out int componentBytesWritten))
+            {
+                bytesWritten = AlgorithmDetails.MLDsaAlgorithm.PublicKeySizeInBytes + componentBytesWritten;
+                return true;
+            }
+
+            bytesWritten = 0;
+            return false;
+        }
+
+        protected override bool TryExportCompositeMLDsaPrivateKeyCore(Span<byte> destination, out int bytesWritten)
+        {
+            // draft-ietf-lamps-pq-composite-sigs-latest (June 20, 2025), 5.2
+            //  1.  Combine and output the encoded private key
+            //
+            //      output mldsaSeed || tradSK
+
+            try
+            {
+                _mldsa.ExportMLDsaPrivateSeed(destination.Slice(0, AlgorithmDetails.MLDsaAlgorithm.PrivateSeedSizeInBytes));
+
+                if (_componentAlgorithm.TryExportPrivateKey(destination.Slice(AlgorithmDetails.MLDsaAlgorithm.PrivateSeedSizeInBytes), out int componentBytesWritten))
+                {
+                    bytesWritten = AlgorithmDetails.MLDsaAlgorithm.PrivateSeedSizeInBytes + componentBytesWritten;
+                    return true;
+                }
+
+                bytesWritten = 0;
+                return false;
+            }
+            catch (CryptographicException)
+            {
+                CryptographicOperations.ZeroMemory(destination);
+                throw;
+            }
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                _mldsa?.Dispose();
+                _mldsa = null!;
+
+                _componentAlgorithm?.Dispose();
+                _componentAlgorithm = null!;
+            }
+
+            base.Dispose(disposing);
+        }
+
+        private static byte[] GetMessageRepresentative(
+            AlgorithmMetadata metadata,
+            ReadOnlySpan<byte> context,
+            ReadOnlySpan<byte> r,
+            ReadOnlySpan<byte> message)
+        {
+            Debug.Assert(r.Length is CompositeMLDsaAlgorithm.RandomizerSizeInBytes);
+
+            // M' = Prefix || Domain || len(ctx) || ctx || r || PH( M )
+
+            using (IncrementalHash hash = IncrementalHash.CreateHash(metadata.HashAlgorithmName))
+            {
+                int length = checked(MessageRepresentativePrefix.Length +   // Prefix
+                                     metadata.DomainSeparator.Length +      // Domain
+                                     1 +                                    // len(ctx)
+                                     context.Length +                       // ctx
+                                     r.Length +                             // r
+#if NET
+                                     hash.HashLengthInBytes);               // PH( M )
+#else
+                                     hash.GetHashLengthInBytes());          // PH( M )
```

Comment:
We don't do inset alignment like this.  I recommend moving `checked` to the method decl (so all the math is checked) and then writing this as

```C#
int length =
    MessageRepresentativePrefix.Length +
    metadata.DomainSeparator.Length +
    ...;
```

You can keep the comments on the end, and aligned, I guess, but to me they're redundant with the M' construction comment.